mainauto.c

```diff
@@ -0,0 +1,39 @@
+#include <stdio.h>
+#include "display.h"
+#include "mapio.h"
+#include "datastrc.h"
+#include "control.h"
+
+int main(int argc, char &argv){
+
+allocate_memory(*gmdt);
```

Comment:
You don't have to call the memory allocation functions. My `read_data()` function already does use `allocate_memory() `in it, and `allocate_pngns()` will be done in the functions belonging to the control.c file.

Also, like I've said above - `&gmdt` here, not `*gmdt`.